### PR TITLE
Feature/time picker

### PR DESCRIPTION
**Pull request by maryafzali24**
## Issue ticket number and link
Drop-7
https://tripleten-apiary.atlassian.net/jira/software/projects/DROP/boards/5?selectedIssue=DROP-7

## Checklist before requesting a review

- [*] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [*] I have made any necessary updates to Storybook to accompany these changes
- [*] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [*] My changes generate no new warnings

### CODE REVIEW

File: src/components/timePicker/timePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import {useForm } from "react-hook-form";
+
+type TimePickerProps = {
+  label?: string;
+}
+const TimePicker = ({ label }: TimePickerProps) => {
+  const { register, formState: { errors } } = useForm();
+  return (
+    <div className="w-346 h-99">
```
Comment (Dillona25) [inline src/components/timePicker/timePicker.tsx:9]: These styles are not correct. With Tailwind, in order to add custom width and height like you are trying you would need to do w-[346px] and h-[99px]. I wouldn't recommend adding the width and height here but rather letting it adjust itself with the widths and heights of the inside components.

File: src/components/timePicker/timePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <label className="text-base leading-[19.2px] font-rubik mb-4">{label}</label>
```
Comment (Dillona25) [inline src/components/timePicker/timePicker.tsx:10]: I wouldnt do ```19.2px``` I would just do ```19px``` this is usually a designer error when they have a weird decimal number.

File: src/components/timePicker/timePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    <div className="h-76 bg-gray-100 rounded-lg flex items-center p-4">
+     <input type="number"
+  {...register('hour', {
+    required: true,
+    min: 0,
+    max: 12,
+    pattern: /^(0?[0-9]|1[0-2])$/,
+})}
+  className="w-60 h-60 border border-solid border-[1px] border-gray-400 bg-gray-100 rounded-lg text-center" />
```
Comment (Dillona25) [inline src/components/timePicker/timePicker.tsx:19]: Adjust dimensions -- already did in call

File: src/components/timePicker/timePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      <span className="ml-[12.5px]">:</span>
+      <input type="number"
+  {...register('minutes', {
+    required: true,
+    min: 0.0,
+    max: 59,
+    pattern: /^(0?[0-9]|[1-5][0-9]|60)$/,
+  })} className="w-60 h-60 border border-solid border-[1px] border-gray-400 bg-gray-100 rounded-lg text-center mx-[12.5px]" />
```
Comment (Dillona25) [inline src/components/timePicker/timePicker.tsx:27]: Adjust dimensions -- already did in call

File: src/components/timePicker/timePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ 
+        <select
+          {...register('period', {required: true})}
+          className="w-60 h-60 border border-solid border-[1px] border-gray-400 bg-gray-100 rounded-lg mx-[1px] text-center"
```
Comment (Dillona25) [inline src/components/timePicker/timePicker.tsx:31]: Adjust dimensions -- already did in call

File: src/components/timePicker/timePicker.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        >
+          <option value="AM">AM</option>
```
Comment (Dillona25) [inline src/components/timePicker/timePicker.tsx:33]: Add ```font-text``` wherever there is text

**Review by Dillona25 — COMMENTED** [review]
The logic here mall looks great. Just some fixes that are needed to the styles. Also I would change the input types to ```strings``` personally, but if you would like to clarify with Josh you can as well.  Once all is done you should have a component like this in StoryBook

Screenshot 2024-04-04 at 6 08 43 PM

**Review by Dillona25 — CHANGES_REQUESTED** [review]
See attached comments

File: src/components/timePicker/timePicker.tsx (hunk 1 above)
Comment (maryafzali24) [reply src/components/timePicker/timePicker.tsx:9]: I deleted the entire className on the parent component.

File: src/components/timePicker/timePicker.tsx (hunk 2 above)
Comment (maryafzali24) [reply src/components/timePicker/timePicker.tsx:10]: Fixed that.

File: src/components/timePicker/timePicker.tsx (hunk 6 above)
Comment (maryafzali24) [reply src/components/timePicker/timePicker.tsx:33]: Added font-text to each text.

File: src/components/timePicker/timePicker.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+type TimePickerProps = {
+  label?: string;
+}
+const TimePicker = ({ label }: TimePickerProps) => {
+  const { register, formState: { errors } } = useForm();
+  return (
+    <div className="w-346 h-99">
+    <label className="text-base leading-[19.2px] font-rubik mb-4">{label}</label>
+    <div className="h-76 bg-gray-100 rounded-lg flex items-center p-4">
+     <input type="number"
```
[the remark this answers is not in the document]
Comment (maryafzali24) [reply src/components/timePicker/timePicker.tsx:12]: Sure, That also solved text alignment. Now number entered by the user is centered within the input field.

**Pull request merged**